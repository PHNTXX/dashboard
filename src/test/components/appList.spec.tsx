import { render } from "@testing-library/react";
import AppList, { IAppListProps } from "../../components/appList";

const props: IAppListProps = {
  categories: [
    {
      name: "Category Test",
      items: [
        {
          name: "App Test",
          icon: "bug_report",
          url: "#",
          displayURL: "test",
          newTab: false,
        },
        {
          name: "App Test",
          icon: "bug_report",
          url: "#",
          displayURL: "test",
          newTab: false,
        },
      ],
    },
  ],
  apps: [
    {
      name: "App Test",
      icon: "bug_report",
      url: "#",
      displayURL: "test",
      newTab: false,
    },
  ],
};

describe("appList.tsx", () => {
  it("Tests AppList rendering with categories and apps", () => {
    const { asFragment } = render(
      <AppList categories={props.categories} apps={props.apps} />,
    );

    expect(asFragment).toMatchSnapshot();
  });

  it("Tests AppList rendering with categories", () => {
    const { asFragment } = render(<AppList categories={props.categories} />);
    expect(asFragment).toMatchSnapshot();
  });

  it("Tests AppList rendering with apps", () => {
    const { asFragment } = render(<AppList apps={props.apps} />);
    expect(asFragment).toMatchSnapshot();
  });

  it("Tests AppList rendering with neither", () => {
    const { asFragment } = render(<AppList />);
    expect(asFragment).toMatchSnapshot();
  });
});
