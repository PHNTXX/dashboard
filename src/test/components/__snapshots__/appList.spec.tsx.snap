// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`appList.tsx Tests AppList rendering with apps 1`] = `[Function]`;

exports[`appList.tsx Tests AppList rendering with categories 1`] = `[Function]`;

exports[`appList.tsx Tests AppList rendering with categories and apps 1`] = `[Function]`;

exports[`appList.tsx Tests AppList rendering with neither 1`] = `[Function]`;
